### PR TITLE
Rename ``IsBound`` to ``HasCapability``

### PR DESCRIPTION
**Pull request by expertdicer**
## Description

closes: #828

### Commit Message / Changelog Entry

```bash
imp(api!): rename `IsBound` to `HasCapability` for IBC application modules
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by crodriguezvega — COMMENTED** [review]
Thanks a lot, @expertdicer! Code changes look good to me. Could you please also update the documentation. There's, I think, a few places:  [here](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/apps.md), [here](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/apps/bindports.md) and [here](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/apps/keeper.md).

Also wondering if the `HasCapability` function is expected to be used by users of ibc-go or if we could make it unexported... @colin-axner?

File: modules/apps/27-interchain-accounts/controller/keeper/keeper.go [hunk 1]
```diff
@@ -98,8 +98,8 @@ func (k Keeper) BindPort(ctx sdk.Context, portID string) *capabilitytypes.Capabi
 	return k.portKeeper.BindPort(ctx, portID)
 }
 
-// IsBound checks if the interchain account controller module is already bound to the desired port
-func (k Keeper) IsBound(ctx sdk.Context, portID string) bool {
+// HasCapability checks if the interchain account controller module is already bound to the desired port
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/keeper.go:101]:
```suggestion
// HasCapability checks if the interchain account controller module owns the port capability for the desired port
```

File: modules/apps/27-interchain-accounts/host/keeper/keeper.go [hunk 2]
```diff
@@ -78,8 +78,8 @@ func (k Keeper) BindPort(ctx sdk.Context, portID string) *capabilitytypes.Capabi
 	return k.portKeeper.BindPort(ctx, portID)
 }
 
-// IsBound checks if the interchain account host module is already bound to the desired port
-func (k Keeper) IsBound(ctx sdk.Context, portID string) bool {
+// HasCapability checks if the interchain account host module is already bound to the desired port
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/keeper/keeper.go:81]:
```suggestion
// HasCapability checks if the interchain account host module owns the port capability for the desired port
```

File: modules/apps/transfer/keeper/keeper.go [hunk 3]
```diff
@@ -64,8 +64,8 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", "x/"+exported.ModuleName+"-"+types.ModuleName)
 }
 
-// IsBound checks if the transfer module is already bound to the desired port
-func (k Keeper) IsBound(ctx sdk.Context, portID string) bool {
+// HasCapability checks if the transfer module is already bound to the desired port
```
Comment (colin-axner) [inline modules/apps/transfer/keeper/keeper.go:67]:
```suggestion
// HasCapability checks if the transfer module owns the port capability for the desired port
```

**Review by colin-axner — APPROVED** [review]
Nice one! Thank you :)

**Comment by colin-axner** [comment]
> Also wondering if the HasCapability function is expected to be used by users of ibc-go or if we could make it unexported... @colin-axner?

Great consideration, we can make it private! External callers should not need to know which application owns the capability, but they can ask the capability module or 05-port if desired, not the application itself

**Review by damiannolan — APPROVED** [review]
Thanks 🙏

**Comment by crodriguezvega** [comment]
> Great consideration, we can make it private! External callers should not need to know which application owns the capability, but they can ask the capability module or 05-port if desired, not the application itself

Cool! I guess we could do that in a follow-up PR, but considering that it's a small modification to this PR, maybe we can also include that here? @expertdicer would you be able to do that?

**Comment by expertdicer** [comment]
> > Great consideration, we can make it private! External callers should not need to know which application owns the capability, but they can ask the capability module or 05-port if desired, not the application itself
> 
> Cool! I guess we could do that in a follow-up PR, but considering that it's a small modification to this PR, maybe we can also include that here? @expertdicer would you be able to do that?

Yes. I will do that.

**Comment by colin-axner** [comment]
I guess making `HasCapability` private will be in a followup pr?

**Comment by damiannolan** [comment]
> I guess making `HasCapability` private will be in a followup pr?

my bad, I was a little too eager with that automerge label, I think a follow up is totally fine

**Comment by expertdicer** [comment]
> I guess making `HasCapability` private will be in a followup pr?

Sorry. I was too busy. You can assign me on this one

**Pull request merged**